app/views/curation_concerns/file_sets/upload/_form_fields.html.erb

```diff
@@ -2,7 +2,7 @@
       <%= hidden_field_tag(:total_upload_size, 0) %>
       <%= hidden_field_tag(:relative_path) %>
       <%= hidden_field_tag(:upload_set_id, @upload_set_id) %>
-      <%= hidden_field_tag(:parent_id, @work_id) %>
+      <%= hidden_field_tag(:parent_id, @presenter.id) %>
```

Comment:
The reason `@work_id` is required is because sometimes this view is rendered as part of Generic Work Show page and sometimes standalone (e.g. when adding files without works.) 

I've set the code to use `@work_id` again (rather than the presenter) but made sure the variable is set.